node/src/main/scala/coop/rchain/node/proposescript/ProposeConfig.scala

```diff
@@ -0,0 +1,132 @@
+package coop.rchain.node.proposescript
+
+import cats.syntax.all._
+import cats.effect.{Concurrent, Sync}
+import coop.rchain.monix.Monixable
+import coop.rchain.shared.Log
+import fs2.Stream
+import retry.{retryingOnFailures, RetryDetails, RetryPolicies, Sleep}
+
+import scala.concurrent.duration.FiniteDuration
+
+final case class ProposeConfig(
+    leader: ValidatorProposeInfo,
+    followers: List[ValidatorProposeInfo],
+    checkInterval: FiniteDuration,
+    proposeInterval: FiniteDuration
+) {
```

Comment:
Classes used for configuration (only constructor fields) should be separated from classes used to define operations.

Functions here are the essence of leaderful proposer so makes sense to call it something like `LeaderfulProposer[F[_]: ...]`. Constructor parameters defines what is necessary to run it.